### PR TITLE
Feature/include/session on registration

### PR DESCRIPTION
**Pull request by Pcushing**

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -11,16 +11,12 @@ def create
       @user = existing_user ## Lead that hasnt created their own password
     end
     @user.assign_attributes(user_params)
-    @user.employer = Organization.where(
-      title: organization_params[:organization_title]
-    ).first_or_initialize
-    @user.role = :basic
-
+    @user.employer = define_employer
+    @user.role = :basic unless @user.role.eql?(:admin)
```
Comment (Pcushing) [inline app/controllers/api/v1/users_controller.rb]: Tried to make this a default param in the model but realized we use it to confirm switches from lead to basic. I slipped this in bc it actually messes up inviting admins as it was converting them to basic, too.

File: spec/requests/api/v1/users_spec.rb [hunk 2]
```diff
@@ -73,9 +87,6 @@
           expect(json["data"]["attributes"]["role"]).to eq "basic"
           expect(json["data"]["relationships"]["employer"]["data"]["id"]).to eq organization.id.to_s
           expect(json["data"]["relationships"]["employer"]["data"]["type"]).to eq "organization"
-          expect(json["included"][0]["id"]).to eq organization.id.to_s
```
Comment (Pcushing) [inline spec/requests/api/v1/users_spec.rb]: All of these `json["included"][0]`  checks are just invalid now that sessions are in there. It didnt seem to make sense to replace as they're covered in the relationships already.

File: app/controllers/api/v1/users_controller.rb (hunk 1 above)
Comment (littleforest) [reply app/controllers/api/v1/users_controller.rb]: Do you think we still need the lead/basic distinction? I had set it up originally so we could identify users who needed to set a password, but we are using the `joined?` method instead for that.  Not sure if it is still helpful to have these separate roles?

If we leave it, then line 15 should be:

```
@user.role = :basic unless @user.admin?
```

File: spec/requests/api/v1/users_spec.rb [hunk 3]
```diff
@@ -19,6 +19,21 @@
...
+      it "create a client_session record" do
+        expect{
+          post path, params: params
+        }.to change(ClientSession, :count).by(1)
+      end
+
+      it "returns session data with access token" do
+        post path, params: params
+        expect(response).to have_http_status(:created)
+        expect(json["data"]["relationships"]["sessions"]["data"][0]["id"]).to be
```
Comment (littleforest) [inline spec/requests/api/v1/users_spec.rb]: missing what it should equal -- why is this passing?

File: spec/requests/api/v1/users_spec.rb [hunk 4]
```diff
@@ -19,6 +19,21 @@
...
+        expect(json["data"]["relationships"]["sessions"]["data"][0]["type"]).to eq "session"
+        expect(json["meta"]["access_token"]).to be
```
Comment (littleforest) [inline spec/requests/api/v1/users_spec.rb]: missing what it should equal -- why is this passing?

File: spec/requests/api/v1/users_spec.rb (hunk 2 above)
Comment (littleforest) [reply spec/requests/api/v1/users_spec.rb]: oh yeah, I started replacing those with: 

```
 expect(json["included"]).to include(a_hash_including("type" => "work_process", "id" => oswp.id.to_s))
```

Can you add that in so we at least have a check that data we want in the `included` array is there?

File: spec/requests/api/v1/users_spec.rb [hunk 5]
```diff
@@ -115,9 +126,6 @@
           expect(json["data"]["attributes"]["role"]).to eq "basic"
           expect(json["data"]["relationships"]["employer"]["data"]["id"]).to eq organization.id.to_s
           expect(json["data"]["relationships"]["employer"]["data"]["type"]).to eq "organization"
-          expect(json["included"][0]["id"]).to eq organization.id.to_s
-          expect(json["included"][0]["type"]).to eq "organization"
-          expect(json["included"][0]["attributes"]["title"]).to eq "Acme Computing"
```
Comment (littleforest) [inline spec/requests/api/v1/users_spec.rb]: Replace with:

```
expect(json["included"]).to include(a_hash_including("type" => "organization", "id" => organization.id.to_s))
```

File: app/controllers/api/v1/users_controller.rb (hunk 1 above)
Comment (Pcushing) [reply app/controllers/api/v1/users_controller.rb]: Let's leave it as-is for now. We can rip it up later.

File: spec/requests/api/v1/users_spec.rb (hunk 3 above)
Comment (Pcushing) [reply spec/requests/api/v1/users_spec.rb]: @littleforest it should just the opposite of `.to_not be_nil`. I wasnt testing for a certain value, just that it is returning one.

File: spec/requests/api/v1/users_spec.rb (hunk 4 above)
Comment (Pcushing) [reply spec/requests/api/v1/users_spec.rb]: @littleforest it should just the opposite of `.to_not be_nil`. I wasnt testing for a certain value, just that it is returning one.

File: spec/requests/api/v1/users_spec.rb (hunk 3 above)
Comment (littleforest) [reply spec/requests/api/v1/users_spec.rb]: Oh cool, I had no idea you could do that.

**Comment by Pcushing** [comment]
@littleforest  should be good-to-go once the specs clear.

**Pull request merged**